lib/path/default.nix

```diff
@@ -94,6 +95,47 @@ let
 
 in /* No rec! Add dependencies on this file at the top. */ {
 
+  /* Append a subpath string to an absolute path value, the result is an absolute path value.
+
+    Like `path + ("/" + string)` but safer, because it errors for potentially unexpected results.
+    More specifically, it checks that the first argument is a path value,
+    and that the second argument is a valid subpath string, see `lib.path.subpath.isValid`.
```

Comment:
Add a link to the manual section so it can be navigated to easily.

Reply:
Unfortunately there's no good way to link to it, markdown syntax here doesn't get rendered. I guess I could dig into making that work